Src/FluentAssertions/Specialized/DelegateAssertionsBase.cs

```diff
@@ -49,6 +50,30 @@ protected ExceptionAssertions<TException> ThrowInternal<TException>(Exception ex
         return new ExceptionAssertions<TException>(expectedExceptions);
     }
 
+    private protected ExceptionAssertions<TException> ThrowInternal<TException>(
+        Exception exception, TimeSpan timeSpan, string because, object[] becauseArgs)
+        where TException : Exception
+    {
+        TException[] expectedExceptions = extractor.OfType<TException>(exception).ToArray();
+
+        Execute.Assertion
+            .BecauseOf(because, becauseArgs)
+            .WithExpectation("Expected a <{0}> to be thrown within {1}{reason}, ",
+                typeof(TException), timeSpan)
+            .ForCondition(exception is not null)
+            .FailWith("but no exception was thrown.")
+            .Then
+            .ForCondition(expectedExceptions.Any())
```

Comment:
We're missing a test where the an exception is thrown within the expected timeout, but of an unexpected type.

Reply:
done